lib/requestPolicy.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { HttpRequest } from "./httpRequest";
+import { HttpResponse } from "./httpResponse";
+import { RequestPolicyOptions } from "./requestPolicyOptions";
+import { HttpPipelineLogLevel } from "./httpPipelineLogLevel";
+
+/**
+ * Uses the decorator pattern to add custom behavior when an HTTP request is made.
+ * e.g. add header, user agent, timeout, retry, etc.
+ */
+export interface RequestPolicy {
+    /**
+     * Send the provided HttpRequest request.
+     * @param request The HTTP request to send.
+     * @return A Promise that resolves to the HttpResponse from the targeted server.
+     */
+    send(request: HttpRequest): Promise<HttpResponse>;
```

Comment:
I guess I'd advocate for making this a type alias for a function type, or just making this an invokable member (remove the `send` and leave everything else).

Reply:
Yeah, I'll change this to be an invokable member. That'll be better for BaseRequestPolicy.

Reply:
I talked about my experience with this in a previous comment. Making this invokable causes problems in BaseRequestPolicy (invokable functions can't be marked as abstract in an abstract base class). I'm going to leave this as a named function.